debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/AbstractReplicationMessageColumn.java

```diff
@@ -153,7 +157,7 @@ public boolean isOptional() {
     private final String columnName;
     private final String typeWithModifiers;
     private final boolean optional;
-    private TypeMetadata typeMetadata;
+    private TypeMetadataImpl typeMetadata;
```

Comment:
Why implementation and not the interface `TypeMetadata`? Tha same applies for few other places.

Reply:
It's needed because there's some stuff like `getFullType()` and `getBaseType()` which I didn't want to expose on this interface. I'd suggest to go with it for now, we can revisit and further clean up this once we've removed support for the old plug-in versions, i.e. once we're sure we'll always receive the type info as part of the messages.